fix: downgrade react version to 17

matijagaspar opened the pull request:
Fixes #121 
Fixes #122

Downgrading to react 17 to match the react version of GraphiQL

simoneb reviewed (changes requested):
dependabot will bump these again

matijagaspar commented:
> dependabot will bump these again

Added [graphiqlReactVersion.test.js](https://github.com/nearform/mercurius-federation-info-graphiql-plugin/pull/123/files#diff-2b97bdbe0cac877f722ab3b725a3a5eb404dbc4cd77761beccaf3f6082fc5018) to check version compatibility between graphiql and the plugin.

Also added ignore to the [dependabot.yml](https://github.com/nearform/mercurius-federation-info-graphiql-plugin/pull/123/files#diff-dd4fbda47e51f1e35defb9275a9cd9c212ecde0b870cba89ddaaae65c5f3cd28). Dependabot should never bump the major version of react, it should come from the GraphiQL.
